src/components/Mapbox.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useRef, useEffect } from 'react';
+import mapboxgl from 'mapbox-gl';
+import { Map, Marker, MapboxOptions, MarkerOptions, LngLatBounds } from 'mapbox-gl';
+import { Result, useSearchState } from '@yext/search-headless-react';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import ReactDOM from 'react-dom';
+
+/**
+ * Interface for customizing the {@link Mapbox} map derived from Mapbox API's Map options.
+ * The "container" field is set internally. The "center" field is offered as a surface level prop
+ * in {@link MapboxProps}.
+ *
+ * @public
+ */
+export type MapboxCustomOptions = Omit<MapboxOptions, 'container' | 'center'>;
+
+/**
+ * Interface for customizing the {@link Mapbox} marker derived from Mapbox API's Marker options.
+ * The "element" field is set internally.
+ *
+ * @public
+ */
+export type MarkerCustomOptions = Omit<MarkerOptions, 'element'>;
+
+/**
+ * An outline to create a custom Marker component.
+ *
+ * @public
+ */
+export type MapPinComponent<TEntity> = (props: { result: Result<TEntity> }) => JSX.Element;
+
+/**
+ * Props for the {@link Mapbox} component.
+ *
+ * @public
+ */
+export interface MapboxProps<TEntity> {
+  mapboxApiKey: string,
+  mapboxOptions?: MapboxCustomOptions,
+  markerOptions?: MarkerCustomOptions,
+  PinComponent?: MapPinComponent<TEntity>,
+  defaultCenter?: Coordinate,
+  getCoordinate: CoordinateGetter<TEntity>
+}
+
+/**
+ * A component that renders a map with markers to show result locations.
+ *
+ * @param props - {@link MapboxProps}
+ * @returns A React element conatining a Google Map
+ *
+ * @public
+ */
+export function Mapbox<TEntity>({
+  mapboxApiKey,
+  mapboxOptions,
+  markerOptions,
+  PinComponent,
+  defaultCenter: center,
+  getCoordinate
+}: MapboxProps<TEntity>) {
+  mapboxgl.accessToken = mapboxApiKey;
```

Comment:
should this be in a useEffect? is this the normal way of setting the mapbox api key?

Reply:
Yeah, its the way that do it in their React [example](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/)-- they don't seem to put in in a hook, but I'll give it some thought

Reply:
Since we only need to set the API key once and not at every render, I wrapped it. Just out of curiosity, is there some sort of overhead/trade-off when using the useEffect hook, or is it still worth using for small things like var assignments?

Reply:
There's some overhead for each new effect you have. In this case that overhead is probably greater than any work you would save by not setting the apiKey extra times (setting property values is extremely fast). This is more on principal of clearly defining side effects. Generally speaking putting effects in useEffect/useLayoutEffect will have more consistent behavior than putting it in the render.